### DISCUSSION TITLE
script.painless.regex.enabled: false

### DISCUSSION THREAD

**Discussion by rijafida** [Security-Onion-Solutions/securityonion#3896] [Unsupported Versions] [answered]
Hi,

Im a newb to security onion 2.3 . the base in centos 7 and i have been trying to edit the elasticsearch.yml file to enable the regex expresion handling in painless.
 following yml files are present in my system
./SecurityOnion/salt/elasticsearch/files/elasticsearch.yml 
I have edited this file to include 
script.painless.regex.enabled : true 
but i am unable to do so 
i have tried editing through following heirarchy
transient 
persistent 
elasticsearch.yml
default 

But to no avail
can anyone pls help me out! I am badly at this one spot

**Comment by weslambert** [comment]
When you say you are "unable to do so" when editing the file, what do you mean?

You should be able to copy `/opt/so/saltstack/default/salt/elasticsearch/files/elasticsearch.yml` to `/opt/so/saltstack/local/salt/elasticsearch/files/elasticsearch.yml` and modify it there.

**Reply by rijafida** [reply]
Thankyou soo much ill give this a try !!!

**Comment by rijafida** [comment, marked as answer]
HI..

i copied the script to elasticsearch.yml file and restarted using sudo so-elastic-restart but the changes dont take effect.

I checked it through query 
GET _cluster/settings?include_defaults=true

**Reply by cm-ops** [reply]
@rijafida as Wes suggested do the following:
1. On the Manager: `cp /opt/so/saltstack/default/salt/elasticsearch/files/elasticsearch.yml /opt/so/saltstack/local/salt/elasticsearch/files/elasticsearch.yml`
2. Add` script.painless.regex.enabled : true` to `/opt/so/saltstack/local/salt/elasticsearch/files/elasticsearch.yml`
3. On your Search node run `so-elasticsearch-restart`
4. Check cluster setting on the Search node with `curl -sk https://localhost:9200/_cluster/settings?include_defaults=true | jq . | grep -C10 script`
5. If you are running a Standalone, do all the above on the Standalone.

You should see the setting in the output of the cluster settings command above.

**Reply by rijafida** [reply]
Thankyou this did the job !!
